src/Bicep.Core/Semantics/DeclarationVisitor.cs

```diff
@@ -206,6 +206,11 @@ private TypeSymbol ResolveProviderSymbol(ProviderDeclarationSyntax syntax)
                 return ErrorType.Create(errorBuilder(DiagnosticBuilder.ForPosition(syntax)));
             }
 
+            else if (typesBaseUri != null && !features.ProviderRegistryEnabled)
```

Comment:
[nit] Just to stick with the existing pattern
```suggestion
            if (typesBaseUri != null && !features.ProviderRegistryEnabled)
```

Reply:
Actually may have to remove this logic, and move it to the DefaultNamespaceProvider, will explain in more detail in the following commit.